Add docs for casting functions (#7793)
cakephp/docs: en/appendices/5-1-upgrade-guide.rst

- ``Arguments::getBooleanOption()`` and ``Arguments::getMultipleOption()`` were added.

Core
----

> - The ``toString``, ``toInt``, ``toBool`` were added. These methods give you
>   a typesafe way to cast request data or other input. Should casting not
>   succeed, ``null`` will be returned.

Reviewer: ```suggestion
- The ``toString``, ``toInt``, ``toBool`` functions were added. They give you
  a typesafe way to cast request data or other input and return ``null`` when conversion fails.
```
